--- Surface_mesh_approximation/include/CGAL/Variational_shape_approximation.h
+++ Surface_mesh_approximation/include/CGAL/Variational_shape_approximation.h
@@ -451,6 +451,15 @@ public:
     return num_added;
   }

+  bool add_to_any_proxy(const face_descriptor f) {
+    std::size_t px_idx = get(m_fproxy_map, f);
+    if (f == m_proxies[px_idx].seed) return false;
+
+    add_one_proxy_at(f);
+
+    return true;
+  }
+
   /*!
    * @brief adds proxies by diffusing fitting error into current partition.
    * Each partition is added with the number of proxies in proportion to its fitting error.
